### PR TITLE
Feature/aa 675 move contracts data conversion

### PR DESCRIPTION
**Pull request by mingazhev**
partially resolves https://github.com/happy-travel/agent-app-project/issues/675

What's done there:
Previously we applied our data processing (currency conversion, markups etc.) to the models from EdoContracts.
This does not fit now, because we need additional info before applying markups.
So I've moved conversion from contracts upper.

To not break other classes, I've created temporary classes-duplicates for common classes, with _New suffix. They will replace the old classes when work on all steps will be done.

### CODE REVIEW

File: Api/Services/Accommodations/Availability/RoomContractSetPriceProcessor_New.cs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            return roomContractSets;
+        }
+
+
+        public static async Task<RoomContractSet> ProcessPrices(RoomContractSet sourceRoomContractSet, PriceProcessFunction priceProcessFunction)
+        {
+            var roomContracts = new List<RoomContract>(sourceRoomContractSet.Rooms.Count);
+            var sourceTotalPrice = sourceRoomContractSet.Rate.FinalPrice;
+            if (sourceTotalPrice.Amount == 0)
+                throw new NotSupportedException("Room contract set price cannot be 0");
```
Comment (kirillta) [inline Api/Services/Accommodations/Availability/RoomContractSetPriceProcessor_New.cs:34]: Is it OK to fail if just one price is fail?

File: Api/Models/Accommodations/Rate.cs [hunk 2]
```diff
@@ -10,12 +10,12 @@ namespace HappyTravel.Edo.Api.Models.Accommodations
     public readonly struct Rate
     {
         [JsonConstructor]
-        public Rate(in MoneyAmount finalPrice, in MoneyAmount gross, List<Discount> discounts,
-            PriceTypes type, string description)
+        public Rate(in MoneyAmount finalPrice, in MoneyAmount gross, List<Discount> discounts = null,
+            PriceTypes type = PriceTypes.Room, string description = "")
```
[the remark this answers is not in the document]
Comment (mingazhev) [reply Api/Models/Accommodations/Rate.cs:14]: Not working 
изображение

File: Api/Services/Accommodations/Availability/RoomContractSetPriceProcessor_New.cs (hunk 1 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Availability/RoomContractSetPriceProcessor_New.cs:34]: I think it is ok, this is a critical error that can indicate that we can't rely on the whole connector result.
Deserialization issue, mapping issue or something like that

File: Api/Services/Accommodations/Availability/RoomContractSetPriceProcessor_New.cs [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            return BuildRoomContractSet(roomContractSet, roomContractSetRate, roomContracts);
+        }
+        
+        
+        private static DailyRate BuildDailyPrice(in DailyRate dailyRate, MoneyAmount roomFinalPrice, MoneyAmount roomGross)
+            => new DailyRate(dailyRate.FromDate, dailyRate.ToDate, roomFinalPrice, roomGross, dailyRate.Type, dailyRate.Description);
+        
+        
+        static RoomContract BuildRoomContracts(in RoomContract room, List<DailyRate> roomPrices, Rate totalPrice)
+            => new RoomContract(room.BoardBasis,
```
Comment (kirillta) [inline Api/Services/Accommodations/Availability/RoomContractSetPriceProcessor_New.cs:122]: You may use a shorter form here as well

File: Api/Services/Accommodations/Availability/RoomContractSetPriceProcessor.cs [hunk 4]
```diff
@@ -64,7 +64,7 @@ public static async Task<RoomContractSet> ProcessPrices(RoomContractSet sourceRo
 
             MoneyAmount ChangeProportionally(MoneyAmount price)
             {
-                var totalPricePercent = price.Amount / sourceTotalPrice.Amount;
+                var totalPricePercent = (price / sourceTotalPrice).Amount;
                 return new MoneyAmount(processedTotalPrice.Amount * totalPricePercent, processedTotalPrice.Currency);
```
Comment (kirillta) [inline Api/Services/Accommodations/Availability/RoomContractSetPriceProcessor.cs:68]: Isn't that just `processedTotalPrice * totalPricePercent`?

File: Api/Models/Accommodations/Rate.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (mingazhev) [reply Api/Models/Accommodations/Rate.cs]: Fixed

File: Api/Services/Accommodations/Availability/RoomContractSetPriceProcessor_New.cs [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+            return BuildRoomContractSet(roomContractSet, roomContractSetRate, roomContracts);
+        }
+        
+        
+        private static DailyRate BuildDailyPrice(in DailyRate dailyRate, MoneyAmount roomFinalPrice, MoneyAmount roomGross)
+            => new (dailyRate.FromDate, dailyRate.ToDate, roomFinalPrice, roomGross, dailyRate.Type, dailyRate.Description);
+        
+        
+        static RoomContract BuildRoomContracts(in RoomContract room, List<DailyRate> roomPrices, Rate totalPrice)
+            => new (room.BoardBasis,
```
Comment (DFSko) [inline Api/Services/Accommodations/Availability/RoomContractSetPriceProcessor_New.cs:122]: 2.2.16 Name primitive parameters explicitly

**Review by DFSko — APPROVED** [review]

**Review by kirillta — APPROVED** [review]

**Pull request merged**